src/lib/util.js

```diff
@@ -13,6 +13,8 @@ const CLIENT_NAME = __NAME__;
 export const CLIENT_VERSION = __VERSION__;
 /* eslint-enable no-undef */
 
+const promiseMap = {};
```

Comment:
Can you use an actual `Map`? This way you get built in methods such as clear for free (which you below)